platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/ReservedSignedState.java

```diff
@@ -116,6 +116,27 @@ public boolean isNotNull() {
         return new ReservedSignedState(signedState, reason);
     }
 
+    /**
+     * Try to get another reservation on the signed state. If the signed state is not closed, then a new reservation
+     * will be returned. If the signed state is closed, then null will be returned.
+     *
+     * @param reason a short description of why this SignedState is being reserved. Each location where a SignedState is
+     *               reserved should attempt to use a unique reason, as this makes debugging reservation bugs easier.
+     * @return a new wrapper around the state that holds a new reservation, or null if the signed state is closed
+     */
+    public @Nullable ReservedSignedState tryGetAndReserve(@NonNull final String reason) {
```

Comment:
I'd suggest adding a short unit test to verify this function's behavior. (OK if it doesn't test race conditions, just that the reservation count is mutated as expected.)

Reply:
added